### DISCUSSION TITLE
How can we focus on an input from within a script?

### DISCUSSION THREAD

**Discussion by shahabbasian** [formkit/formkit#812] [General]
I need to focus on an `<FormKit type="text" />` from within a `<script setup>` block and a `method`

 I would appreciate it if you could guide me.

**Comment by fenilli** [comment]
For now FormKit doesn't have any way to get the input element, but you can still use JS `const el = document.getElementById('elementID')?.focus()` like so.

**Reply by crishoj** [reply]
I wasn't able to focus an input element using plain JS (`document.getElementById('elementID')?.focus()`).

Do you think there is anything in FormKit that could prevent this from working?

**Comment by MichaelSchmidle** [comment]
It'd be great to have the ability to define on form level which field should auto-focus, if any. The result should be the same as with:

```ts
const email = ref<HTMLInputElement | null>(null)
useFocus(email, { initialValue: true })
```

**Comment by fenilli** [comment]
Internally I would guess `useFocus` does the same as `document.getElementById('elementID')?.focus()`, might be able to create a plugin that adds a helper function to fns to be called within the node.
